NLog.Windows.Forms/RichTextBoxLinkLayoutRenderer .cs

```diff
@@ -0,0 +1,83 @@
+﻿using NLog.Config;
+using NLog.LayoutRenderers;
+using NLog.Layouts;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace NLog.Windows.Forms
+{
+    /// <summary>
+    /// Strings rendered with this rendrer would convert to links in the control. <see cref="RichTextBoxTarget.SupportLinks"/>
+    /// </summary>
+    [LayoutRenderer("rtb-link")]
+    public sealed class RichTextBoxLinkLayoutRenderer : LayoutRenderer
+    {
+        /// <summary>
+        /// Inner layout that actually provides text
+        /// </summary>
+        [DefaultParameter]
+        public Layout Inner { get; set; }
+
+        /// <summary>
+        /// Implementation of a <see cref="LayoutRenderer.Append"/>
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="logEvent"></param>
+        protected override void Append(StringBuilder builder, LogEventInfo logEvent)
+        {
+            string msg = this.Inner.Render(logEvent);
+
+            //store new linkInfo to be retreived by RichTextBox
+
+            //TODO: should we synchronize access to logEvent.Properties??
+            LinkInfo linkInfo;
+            object linkInfoObj;
+            if (logEvent.Properties.TryGetValue(LinkInfo.PropertyName, out linkInfoObj))
+            {
+                linkInfo = (LinkInfo)linkInfoObj;
```

Comment:
I'm not sure on this. Explicit cast would at least fail in (unlikely) case somebody have added something else with this key.